Unable to connect to PostgreSQL on Cloud SQL using Cloud SQL R2DBC configuration.

<!--

Thanks for stopping by to let us know something could be better!

**PLEASE READ**: If you have a support contract with Google, please create an 
issue in the [support console](https://cloud.google.com/support/) instead of 
filing on GitHub. This will ensure a timely response.

Please run down the following list and make sure you've tried the usual "quick fixes":

  - Search the issues already opened: https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/issues
  - Check for answers on StackOverflow: https://stackoverflow.com/questions/tagged/google-cloud-sql

If you are still having issues, please include as much information as possible:

--> 

## Bug Description

I am currently trying to set up an r2dbc connection to Cloud SQL from a Google App Engine Spring Boot app using the guide [here](https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/blob/main/docs/r2dbc-postgres.md). After setting up as specified and deploying to App Engine, App Engine throws an error `Caused by: java.lang.IllegalArgumentException: [%s] Cloud SQL connection name is invalid, expected string in the form of "<PROJECT_ID>:<REGION_ID>:<INSTANCE_ID>".`.

What am I missing?

## Example code (or command)

```
                String url;
		if (env.equalsIgnoreCase(Environments.PRODUCTION.getEnvironment()))    //I can confirm this is the section that's executed
		{
		        //GCP Specific connection
			url = "r2dbc:gcp:postgres://" + username + ":" + password + "@" + host + "/" + database;
		}
		else
		{
			url = "r2dbc:postgres://" + username + ":" + password + "@" + host + ":5432/" + database + "?lock_timeout=60s&statement_timeout=5min";
		}

		ConnectionFactory factory = ConnectionFactories.get(url);

		ConnectionPoolConfiguration configuration = ConnectionPoolConfiguration.builder(factory)
			.initialSize(initPoolSize)
			.maxSize(maxPoolSize)
			.maxCreateConnectionTime(Duration.ofSeconds(connectTimeoutSeconds))
			.build();
		return new ConnectionPool(configuration);
```

## App Config
```
datasource:
    host: "<project-id>:europe-west1:<instance-name>"    # real values redacted
    port: 5432
    database: "<db-name>"
    username: ${db_username}
    password: ${db_password}
```

I also tried adding parameters like `?socketFactory=com.google.cloud.sql.postgres.SocketFactory&cloudSqlInstance=<project-id>:europe-west1:<instance-name>` but no change was observed

## Stacktrace
```
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [io.r2dbc.spi.ConnectionFactory]: Factory method 'connectionFactory' threw exception; nested exception is java.lang.IllegalArgumentException: [%s] Cloud SQL connection name is invalid, expected string in the form of "<PROJECT_ID>:<REGION_ID>:<INSTANCE_ID>".
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:185) ~[spring-beans-5.3.15.jar!/:5.3.15]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:653) ~[spring-beans-5.3.15.jar!/:5.3.15]
	... 109 common frames omitted
Caused by: java.lang.IllegalArgumentException: [%s] Cloud SQL connection name is invalid, expected string in the form of "<PROJECT_ID>:<REGION_ID>:<INSTANCE_ID>".
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:144) ~[guava-30.1.1-jre.jar!/:na]
	at com.google.cloud.sql.core.CloudSqlInstance.<init>(CloudSqlInstance.java:138) ~[jdbc-socket-factory-core-1.4.3.jar!/:na]
	at com.google.cloud.sql.core.CoreSocketFactory.lambda$getCloudSqlInstance$0(CoreSocketFactory.java:158) ~[jdbc-socket-factory-core-1.4.3.jar!/:na]
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1705) ~[na:na]
	at com.google.cloud.sql.core.CoreSocketFactory.getCloudSqlInstance(CoreSocketFactory.java:156) ~[jdbc-socket-factory-core-1.4.3.jar!/:na]
	at com.google.cloud.sql.core.CoreSocketFactory.getSslData(CoreSocketFactory.java:259) ~[jdbc-socket-factory-core-1.4.3.jar!/:na]
	at com.google.cloud.sql.core.GcpConnectionFactoryProvider.createFactory(GcpConnectionFactoryProvider.java:117) ~[cloud-sql-connector-r2dbc-core-1.4.3.jar!/:na]
	at com.google.cloud.sql.core.GcpConnectionFactoryProvider.create(GcpConnectionFactoryProvider.java:96) ~[cloud-sql-connector-r2dbc-core-1.4.3.jar!/:na]
	at io.r2dbc.spi.ConnectionFactories.find(ConnectionFactories.java:110) ~[r2dbc-spi-0.9.0.RELEASE.jar!/:na]
	at io.r2dbc.spi.ConnectionFactories.get(ConnectionFactories.java:140) ~[r2dbc-spi-0.9.0.RELEASE.jar!/:na]
	at io.r2dbc.spi.ConnectionFactories.get(ConnectionFactories.java:128) ~[r2dbc-spi-0.9.0.RELEASE.jar!/:na]
	at ke.co.proxyapi.mpesa.transactionsprocessor.config.R2dbcConfig.connectionFactory(R2dbcConfig.java:67) ~[classes!/:1.0.0-SNAPSHOT]
	at ke.co.proxyapi.mpesa.transactionsprocessor.config.R2dbcConfig$$EnhancerBySpringCGLIB$$7d38f98.CGLIB$connectionFactory$0(<generated>) ~[classes!/:1.0.0-SNAPSHOT]
	at ke.co.proxyapi.mpesa.transactionsprocessor.config.R2dbcConfig$$EnhancerBySpringCGLIB$$7d38f98$$FastClassBySpringCGLIB$$f2b73878.invoke(<generated>) ~[classes!/:1.0.0-SNAPSHOT]
	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:244) ~[spring-core-5.3.15.jar!/:5.3.15]
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:331) ~[spring-context-5.3.15.jar!/:5.3.15]
	at ke.co.proxyapi.mpesa.transactionsprocessor.config.R2dbcConfig$$EnhancerBySpringCGLIB$$7d38f98.connectionFactory(<generated>) ~[classes!/:1.0.0-SNAPSHOT]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:566) ~[na:na]
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:154) ~[spring-beans-5.3.15.jar!/:5.3.15]
	... 110 common frames omitted
```

## How to reproduce

  1. Create a PostgreSQL DB on Google Cloud SQL (can be empty)
  2. Set up a Spring Boot R2DBC Application and set up a Google SQL Connection using the instructions at [Github](https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/blob/main/docs/r2dbc-postgres.md)
  3. Authenticate to gcloud
  4. Upload the application to Google App Engine via gcloud
  5. Try to run the application on GAE
  6. Check the logs explorer on console for the status of the application

## Environment

1. Java SDK version: `Java 11`
2. Socket Factory version: `1.4.0`
3. Postgres R2DBC version: `0.8.11.RELEASE`
4. R2DBC Pool version: `0.9.0.RELEASE`

--- a/core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java
+++ b/core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java
@@ -133,12 +133,14 @@ class CloudSqlInstance {
       ListeningScheduledExecutorService executor,
       ListenableFuture<KeyPair> keyPair) {
 
-    this.connectionName = connectionName;
     Matcher matcher = CONNECTION_NAME.matcher(connectionName);
     checkArgument(
         matcher.matches(),
-        "[%s] Cloud SQL connection name is invalid, expected string in the form of"
-            + " \"<PROJECT_ID>:<REGION_ID>:<INSTANCE_ID>\".");
+        String.format(
+            "[%s] Cloud SQL connection name is invalid, expected string in the form of"
+                + " \"<PROJECT_ID>:<REGION_ID>:<INSTANCE_ID>\".",
+            connectionName));
+    this.connectionName = connectionName;
     this.projectId = matcher.group(1);
     this.regionId = matcher.group(3);
     this.instanceId = matcher.group(4);

